torch_geometric/datasets/lrgb.py

```diff
@@ -0,0 +1,330 @@
+    }
+
+    def __init__(self, root: str, name: str, split: str = "train",
+                 transform: Optional[Callable] = None,
+                 pre_transform: Optional[Callable] = None,
+                 pre_filter: Optional[Callable] = None):
+        self.name = name.lower()
+        assert self.name in self.names
+        assert split in ['train', 'val', 'test']
+
+        super().__init__(root, transform, pre_transform, pre_filter)
+        path = osp.join(self.processed_dir, f'{split}.pt')
+        self.data, self.slices = torch.load(path)
+
+    @property
+    def raw_dir(self) -> str:
+        return osp.join(self.root, self.name, 'raw')
+
+    @property
+    def processed_dir(self) -> str:
+        return osp.join(self.root, self.name, 'processed')
+
+    @property
+    def raw_file_names(self) -> List[str]:
+        if self.name.split('-')[1] == 'sp':
+            return ['train.pickle', 'val.pickle', 'test.pickle']
+        else:
+            return ['train.pt', 'val.pt', 'test.pt']
+
+    @property
+    def processed_file_names(self) -> List[str]:
+        return ['train.pt', 'val.pt', 'test.pt']
+
+    def download(self):
+        shutil.rmtree(self.raw_dir)
+        path = download_url(self.urls[self.name], self.root)
+        extract_zip(path, self.root)
+        os.rename(osp.join(self.root, self.dwnld_file_name[self.name]),
+                  self.raw_dir)
+        os.unlink(path)
+
+    def process(self):
+        if self.name.split('-')[1] == 'sp':
+            # PascalVOC-SP and COCO-SP
+            self.process_sp()
+        elif self.name.split('-')[0] == 'peptides':
+            # Peptides-func and Peptides-struct
+            self.process_peptides()
+        else:
+            # PCQM-Contact
+            self.process_pcqm_contact()
+
+    def process_sp(self):
+        for split in ['train', 'val', 'test']:
+            with open(osp.join(self.raw_dir, f'{split}.pickle'), 'rb') as f:
+                graphs = pickle.load(f)
+
+            indices = range(len(graphs))
+
+            pbar = tqdm(total=len(indices))
+            pbar.set_description(f'Processing {split} dataset')
+
+            data_list = []
+            for idx in indices:
+                graph = graphs[idx]
+                """
+                Each `graph` is a tuple (x, edge_attr, edge_index, y)
+                    Shape of x : [num_nodes, 14]
+                    Shape of edge_attr : [num_edges, 1] or [num_edges, 2]
+                    Shape of edge_index : [2, num_edges]
+                    Shape of y : [num_nodes]
+                """
+
+                x = graph[0].to(torch.float)
+                edge_attr = graph[1].to(torch.float)
+                edge_index = graph[2]
+                y = torch.LongTensor(graph[3])
+
+                if self.name == 'coco-sp':
+                    # Label remapping for coco-sp.
+                    # See self.label_remap_coco() func
+                    label_map = self.label_remap_coco()
```

Comment:
This can be simplified with `sklearning.preprocessing.LabelEncoder`, e.g., 
```python
from sklearn.preprocessing import LabelEncoder
y = LabelEncoder().fit_transform(graph[3])
y = torch.LongTensor(y)
```